app/models/spree_avatax/sales_invoice.rb

```diff
@@ -25,6 +25,11 @@ def generate(order)
 
       return if order.completed? || !SpreeAvatax::Shared.taxable_order?(order)
 
+      taxable_records = order.line_items + order.shipments
+      taxable_records.each do |taxable_record|
+        taxable_record.update_column(:pre_tax_amount, taxable_record.discounted_amount.round(2))
+      end
```

Comment:
I'm a little freaked out by this. Invoice generation is causing line item amount and shipment to be updated. That's not the sort of behaviour that I would expect when generating and invoice.